src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int INITIAL_CAPACITY = 10;
+    public static final String NO_SUCH_ELEMENT_MESSAGE = "There is no such element in the list - ";
+    public static final String OUT_OF_BOUNDS_MESSAGE = "Index out of bounds";
+
+    private Object[] storage;
+    private int size = 0;
+
+    public ArrayList() {
+        storage = new Object[INITIAL_CAPACITY];
+    }
+
+    public ArrayList(int capacity) {
+        if (capacity >= 0) {
+            storage = new Object[capacity];
+        } else {
+            throw new ArrayListIndexOutOfBoundsException(OUT_OF_BOUNDS_MESSAGE);
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        storage[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        ensureCapacity();
+        ensurePosition(index - 1);
+        Object[] temp = new Object[storage.length];
+        System.arraycopy(storage, 0, temp, 0, index);
+        temp[index] = value;
+        System.arraycopy(storage, index, temp, index + 1, size - index);
+        storage = temp;
+        size++;
```

Comment:
maybe in we can use add(value) method if index one more then size

i mean 
if (index - 1 == size) {
add(value);
return:
} else {
your code